R/estimateParams.R

```diff
@@ -53,3 +53,97 @@ prepareData <- function(data) {
 
   as.data.frame(dataNew[, -which(names(dataNew) == "time")], row.names = seq_len(nrow(dataNew)))
 }
+
+
+# transition <- weibull_transition(h01 = 1.5, h02 = 0.5, h12 = 1.1, p01 = 0.4, p02 = 0.5, p12 = 2.4)
+# trial <- getOneClinicalTrial(nPat = c(10000),
+#                             transitionByArm = list(transition),
+#                             dropout = list(rate = 0.3, time = 1),
+#                             accrual = list(param = "intensity", value = 100))
+
+# estimateParams(trial, "Weibull")
+
+NegLogLik <- function(transition, data) {
+  UseMethod(NegLogLik)
+}
+
+NegLogLik.ExponentialTransition <- function(transition, data) {
+  with(data, -sum(log(haz(transition, exit, trans)^status * SurvTrans(transition, exit, trans) / SurvTrans(transition, entry, trans))))
+}
+
+NegLogLik.WeibullTransition <- function(transition, data) {
+  with(data, -sum(log(haz(transition, exit, trans)^status * SurvTrans(transition, exit, trans) / SurvTrans(transition, entry, trans))))
+}
+
+haz <- function(transition, t, trans = c(1, 2, 3)) {
+  trans <- match.arg(trans)
+  UseMethod(haz)
+}
+
+haz.ExponentialTransition <- function(transition, t, trans = c(1, 2, 3)) {
+  # params (in this order): h01, h02, h12
+  params <- unlist(transition$hazards)
+  exp(-params[trans] * t)
+}
+
+haz.WeibullTransition <- function(transition, t, trans = c(1, 2, 3)) {
+  # params (in this order): h01, h02, h12, p01, p02, p12
+  params <- c(unlist(transition$hazards), unlist(transition$weibull_rates))
+  exp(-params[trans] * t)
+}
+
+SurvTrans <- function(transition, t, trans = c(1, 2, 3)) {
```

Comment:
this we also need, also here I would rename to `survTrans` since it is not a class where we would use upper case